eng/tools/ChangeValidation/Validate.cs

```diff
@@ -0,0 +1,194 @@
+            try
+            {
+                submoduleMessages = SubmoduleValidation.VerifySubModuleFileChanges(changedFiles);
+            }
+            catch (Exception)
+            {
+                submoduleMessages = new List<ProcessingMessage>();
+                AddProcessingMessage(processingMessages, Error("Error: Failed to perform validation on submodule files."));
+            }
+            processingMessages.AddRange(submoduleMessages);
+
+            // Run Exclusion File Validation
+            Console.WriteLine("");
+            Console.WriteLine("");
+            Console.WriteLine("3. Starting Exclusion File Validation...");
+            List<ProcessingMessage> exclusionMessages;
+            try
+            {
+                exclusionMessages = ExclusionFileValidation.VerifyFileExclusions(changedFiles, targetBranch);
+            }
+            catch (Exception)
+            {
+                exclusionMessages = new List<ProcessingMessage>();
+                AddProcessingMessage(processingMessages, Error("Error: Failed to perform validation on exclusion files."));
+            }
+            processingMessages.AddRange(exclusionMessages);
+        }
+        catch (Exception)
+        {
+            Console.WriteLine($"An unexpected error occurred during the validation process.");
+            return 1;
+        }
+
+        var warnings = processingMessages
+            .Where(m => m.WarningLevel == WarningLevel.Warning)
+            .ToList();
+
+        var errors = processingMessages
+            .Where(m => m.WarningLevel == WarningLevel.Error)
+            .ToList();
+
+        if (errors.Any())
+        {
+            Console.WriteLine($"VMR Change Validation failed with {errors.Count} error(s) and {warnings.Count} warning(s).");
+            return 1;
+        }
+        else
+        {
+            Console.WriteLine($"VMR Change Validation completed successfully with {warnings.Count} warning(s).");
+            return 0;
+        }
+    }
+
+
+
+    internal static string RunGitCommand(string arguments)
+    {
+        var process = new Process
+        {
+            StartInfo = new ProcessStartInfo
+            {
+                FileName = "git",
+                Arguments = arguments,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                UseShellExecute = false,
+                CreateNoWindow = true
+            }
+        };
+
+        process.Start();
+        string output = process.StandardOutput.ReadToEnd();
+        string err = process.StandardError.ReadToEnd();
+        process.WaitForExit();
+
+        if (process.ExitCode != 0)
+        {
+            throw new Exception($"Git command failed: {arguments}\n{err}");
+        }
+
+        return output;
+    }
+
+    internal static void AddProcessingMessage(List<ProcessingMessage> processingMessages, ProcessingMessage processingMessage)
```

Comment:
Are we buffering the messages just for the sake of counting them at the end or is there another reason?

I think it'd be just better to output them right away and if we need the count, increment some counter then?